src/hotspot/share/memory/metaspace.cpp

```diff
@@ -992,6 +992,7 @@ bool Metaspace::is_in_shared_metaspace(const void* ptr) {
 
 // Returns true if pointer points into one of the non-class-space metaspace regions.
 bool Metaspace::is_in_nonclass_metaspace(const void* ptr) {
-  return VirtualSpaceList::vslist_nonclass()->contains((MetaWord*)ptr);
+  VirtualSpaceList* list = VirtualSpaceList::vslist_nonclass();
+  return (list != nullptr) && list->contains((MetaWord*) ptr);
```

Comment:
I'm not sure about this. This, and a number of other related functions nearby,
all seem like they should have Metaspace::initialized() as a precondition. And
the call to Metaspace::contains in os::print_location should be conditional on
Metaspace::initialized().  (There may be other "is initialized" checks that
should be made by (and offered to) os::print_location, but that's beyond the
scope of this change.)

Reply:
FWIW, we had a similar discussion in #19435.

Reply:
All of these methods are public. I think that it seems sane to assume that they are the boundary for checking whether Metaspace is initialized before doing something.

Reply:
All right, I think checking for `Metaspace::initialized()` in `os::print_location` is more in line with other changes, notably [JDK-8333290](https://bugs.openjdk.org/browse/JDK-8333290). This is Kim's suggestion, I think. So I did just that.